### PR TITLE
Add additional information about lock inheritance

### PR DESCRIPTION
**Pull request by PypWalters**
## Description
Adds clarity to the documentation concern mentioned here: https://github.com/WordPress/gutenberg/issues/18711

### CODE REVIEW

File: docs/designers-developers/developers/block-api/block-templates.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Templates
 
-A block template is defined as a list of block items. Such blocks can have predefined attributes, placeholder content, and be static or dynamic. Block templates allow to specify a default initial state for an editor session.
+A block template is defined as a list of block items. Such blocks can have predefined attributes, placeholder content, and be static or dynamic. Block templates allow for specifying a default initial state for an editor session.
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/block-api/block-templates.md:3]:
```suggestion
A block template is defined as a list of block items. Such blocks can have predefined attributes, placeholder content, and be static or dynamic. Block templates allow specifying a default initial state for an editor session.
```

File: docs/designers-developers/developers/block-api/block-templates.md [hunk 2]
```diff
@@ -110,6 +110,8 @@ add_action( 'init', 'myplugin_register_template' );
 - `all` — prevents all operations. It is not possible to insert new blocks, move existing blocks, or delete blocks.
 - `insert` — prevents inserting or removing blocks, but allows moving existing blocks.
 
+Lock settings can be inherited by InnerBlocks. If `templateLock` is not set in an InnerBlocks area: the locking of the parent InnerBlocks area is used. If the block is a top level block: the locking of the Custom Post Type is used.
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/block-api/block-templates.md:113]:
```suggestion
Lock settings can be inherited by InnerBlocks. If `templateLock` is not set in an InnerBlocks area, the locking of the parent InnerBlocks area is used. If the block is a top level block, the locking configuration of the current post type is used.
```

**Review by chrisvanpatten — DISMISSED** [review]
A few small grammatical/clarity suggestions, but overall this is great.

**Review by youknowriad — APPROVED** [review]
Thanks 👍

**Pull request merged**